ompi/errhandler/errhandler.c

```diff
@@ -222,6 +231,165 @@ ompi_errhandler_t *ompi_errhandler_create(ompi_errhandler_type_t object_type,
+    for( i = 0; i < max_num_comm; ++i ) {
+        comm = (ompi_communicator_t *)opal_pointer_array_get_item(&ompi_mpi_communicators, i);
+        if( NULL == comm ) {
+            continue;
+        }
+
+        /* Look in both the local and remote group for this process */
+        proc_rank = ompi_group_proc_lookup_rank(comm->c_local_group, ompi_proc);
+        remote = false;
+        if( (proc_rank < 0) && (comm->c_local_group != comm->c_remote_group) ) {
+            proc_rank = ompi_group_proc_lookup_rank(comm->c_remote_group, ompi_proc);
+            remote = true;
+        }
+        if( proc_rank < 0 )
+            continue;  /* Not in this communicator, continue */
+
+        /* Notify the communicator to update as necessary */
+        ompi_comm_set_rank_failed(comm, proc_rank, remote);
+
+        if( NULL == group ) {  /* Build a group with the failed process */
+            rc = ompi_group_incl((remote ? comm->c_remote_group : comm->c_local_group),
+                                 1, &proc_rank,
+                                 &group);
+            if(OPAL_UNLIKELY( OMPI_SUCCESS != rc )) goto cleanup;
+        }
+        OPAL_OUTPUT_VERBOSE((10, ompi_ftmpi_output_handle,
+                             "%s ompi: Process %s is in comm (%d) with rank %d. [%s]",
+                             OMPI_NAME_PRINT(OMPI_PROC_MY_NAME),
+                             OMPI_NAME_PRINT(&ompi_proc->super.proc_name),
+                             comm->c_contextid,
+                             proc_rank,
+                             (OMPI_ERRHANDLER_TYPE_PREDEFINED == comm->errhandler_type ? "P" :
+                              (OMPI_ERRHANDLER_TYPE_COMM == comm->errhandler_type ? "C" :
+                               (OMPI_ERRHANDLER_TYPE_WIN == comm->errhandler_type ? "W" :
+                                (OMPI_ERRHANDLER_TYPE_FILE == comm->errhandler_type ? "F" : "U") ) ) )
+                             ));
+    }
+
+    /* Group State:
+     * Add the failed process to the global group of failed processes. */
+    if( group != NULL ) {
+        ompi_group_t *tmp = ompi_group_all_failed_procs;
+        rc = ompi_group_union(tmp, group,
+                              &ompi_group_all_failed_procs);
+        if(OPAL_UNLIKELY( OMPI_SUCCESS != rc )) goto cleanup;
+        OBJ_RELEASE(tmp);
+    }
+
+    /* Point-to-Point:
+     * Let the active request know of the process state change.
+     * The wait function has a check, so all we need to do here is
+     * signal it so it will check again.
+     */
+    wait_sync_global_wakeup(MPI_ERR_PROC_FAILED);
+
+    /* Collectives:
+     * Propagate the error (this has been selected rather than the "roll
+     * forward through errors in collectives" as this is less intrusive to the
+     * code base.) */
+    if( forward ) {
+        /* TODO: this to become redundand when pmix has rbcast */
+        ompi_comm_failure_propagate(&ompi_mpi_comm_world.comm, ompi_proc, status);
+#if 0
```

Comment:
Another #if 0 to remove?

Reply:
Want to keep, will switch when PRTE support complete